### DISCUSSION TITLE
Are there any intentions around supporting RSC, specifically with nextjs and their upcoming server actions?

### DISCUSSION THREAD

**Discussion by remy90** [pmndrs/jotai#2031] [Ideas] [answered]
I've seen a few comments around not keeping state across requests and SSR support
i.e.
https://github.com/pmndrs/jotai/discussions/1848
https://github.com/pmndrs/jotai/issues/340

My question is: are there any intentions to support/integrate with [server actions](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions#action)?  

I appreciate this may be a completely different project to what jotai currently offers as far as scope is concerned, but I'm looking for some idea from the maintainers about the future roadmap for this project in this regard

**Comment by dai-shi** [comment, marked as answer]
My understanding is that it's impossible to deeply integrate client states with nextjs.

That's why I started developing Waku and https://github.com/dai-shi/waku/pull/11 which is completely WIP. I think it will take months or years.

**Reply by remy90** [reply]
Interesting, would I've noticed Waku is a minimalistic framework - do you envisage this as a replacement for nextjs/similar frameworks or something that would sit alongside other large-scale frameworks? I appreciate this may be a false dichotomy but interested to hear more 🍿

**Reply by dai-shi** [reply]
https://github.com/dai-shi/waku/issues/25 Waku's goal is not to replace nextjs/other frameworks. We'll see...
